disco_aws_automation/disco_metanetwork.py

```diff
@@ -236,3 +251,28 @@ def add_sg_rule(self, protocol, ports, sg_source=None, cidr_source=None):
         sg_args["to_port"] = ports[1]
         logging.debug("Adding sg_rule: %s", sg_args)
         self.vpc.vpc.connection.authorize_security_group(**sg_args)
+
+    def ip_by_offset(self, offset):
+        """
+        Pass in +10 and get 10th ip of subnet range
+        Pass in -2 and get 2nd to last ip of subnet
+
+        Returns IpAddress object, usually you'll want
+        to cast this to str.
+        """
+
+        try:
+            offset = int(offset)
+        except ValueError:
+            raise IPRangeError(
+                "Cannot find IP in metanetwork {0} by offset {1}."
+                .format(self.name, offset)
+            )
+
+        subnets = sorted(self.subnet_ip_networks)
```

Comment:
I believe this returns something like `[IPNetwork('10.101.0.0/20'), IPNetwork('10.101.16.0/20'), IPNetwork('10.101.32.0/20'), IPNetwork('10.101.48.0/20')]`

The line below `base_address = subnets[0].first if offset >= 0 else subnets[-1].last`
will return first IP of first subnet if offset >=0, and last IP of last subnet otherwise.

Am I understanding this wrong?

We want first IP of a certain subnet if offset >=0, or last IP of **the same subnet** otherwise.